libs/wingc/src/type_check.rs

```diff
@@ -708,6 +708,18 @@ impl TypeRef {
 		}
 	}
 
+	pub fn collection_inner_type(&self) -> Option<TypeRef> {
+		match **self {
+			Type::Array(t) => Some(t),
+			Type::MutArray(t) => Some(t),
+			Type::Map(t) => Some(t),
+			Type::MutMap(t) => Some(t),
+			Type::Set(t) => Some(t),
+			Type::MutSet(t) => Some(t),
+			_ => None,
+		}
+	}
```

Comment:
What about adding `Type::Optional(t)` as a case? Philosophically, you can think of an optional as a collection with at most one element. (In Rust, the `Option` type is iterable as an example)

Reply:
I am not sure it makes sense for me to treat `Type::Optional` as a collection.